react-components/src/hooks/use3dScenes.tsx

```diff
@@ -12,95 +12,106 @@ import { type Cdf3dRevisionProperties } from './types';
 import { Euler, MathUtils, Matrix4 } from 'three';
 import { CDF_TO_VIEWER_TRANSFORMATION } from '@cognite/reveal';
 
+export type Scene3DModel = {
+  externalId: string;
+  space: string;
+  addModelOptions: AddReveal3DModelOptions[];
+};
+
 export const use3dScenes = (
   userSdk?: CogniteClient
-): UseQueryResult<Record<string, AddReveal3DModelOptions[]>> => {
+): UseQueryResult<Record<string, Scene3DModel>> => {
   const sdk = useSDK(userSdk);
 
   const fdmSdk = useMemo(() => new FdmSDK(sdk), [sdk]);
 
-  const queryFunction: QueryFunction<Record<string, AddReveal3DModelOptions[]>> = async () => {
+  const queryFunction: QueryFunction<Record<string, Scene3DModel>> = async () => {
     const scenesQuery = createGetScenesQuery();
 
     try {
       const scenesQueryResult = await fdmSdk.queryNodesAndEdges(scenesQuery);
 
-      console.log('scenesQueryResult', scenesQueryResult);
-
-      const scenesMap: Record<string, AddReveal3DModelOptions[]> =
-        scenesQueryResult.items.sceneModels.reduce(
-          (acc, item) => {
-            const edge = item as EdgeItem;
-
-            const { externalId } = edge.startNode;
-
-            const properties = Object.values(
-              Object.values(edge.properties)[0] as Record<string, unknown>
-            )[0] as Cdf3dRevisionProperties;
-            const sceneModels = acc[externalId];
+      const sceneList: Scene3DModel[] = [];
+      scenesQueryResult.items.sceneModels.forEach(
+        (item) => {
```

Comment:
Talking about naming things - can we factor out this lambda with a meaningful name so that it's easier to understand what it does? :sweat_smile:

Reply:
(* referencing the lambda starting at the bottom of the block)

Reply:
Fixed in [8a5bc15](https://github.com/cognitedata/reveal/pull/4051/commits/8a5bc15405652031f72a1f75e548e2ccf772e4e3)